js/map.js

```diff
@@ -45,25 +77,63 @@ map.on("load", () => {
     },
   })
 
-  map.addLayer({
-    id: "ihs-borders",
-    type: "line",
-    source: "ihs_rollup_source",
-    "source-layer": "ihs_rollup-2jzfoj",
-    paint: {
-      "line-color": "#000000",
-      "line-width": [
-        "case",
-        ["boolean", ["feature-state", "selected"], false],
-        5,
-        1,
-      ],
-    },
-  })
+  map
+    .addLayer({
+      id: "ihs-borders",
+      type: "line",
+      source: "ihs_rollup_source",
+      "source-layer": "ihs_rollup_categorized-bu6v14",
+      paint: {
+        "line-color": "#000000",
+        "line-width": [
+          "case",
+          ["boolean", ["feature-state", "selected"], false],
+          5,
+          1,
+        ],
+      },
+    })
+    .once("idle", () => {
+      // Once resources are loaded, set options for the searchable select input
+      const allAreasRaw = map.querySourceFeatures("ihs_rollup_source", {
+        sourceLayer: "ihs_rollup_categorized-bu6v14",
+      })
+      let areaOptions = [
+        { label: "City of Chicago", choices: [] },
+        { label: "Suburban Cook County", choices: [] },
+      ]
+
+      // Remove duplicates from raw response
+      allAreasRaw.forEach((currArea) => {
+        for (const distinctArea of allAreas) {
+          if (distinctArea.id == currArea.id) {
+            return
+          }
+        }
+        allAreas.push(currArea)
+      })
+
+      // Populate area options
+      allAreas.forEach((area) => {
+        const currCategory = area.properties.category
+        const groupIndex = currCategory.includes("City") ? 0 : 1
+        areaOptions[groupIndex].choices.push({
+          value: area.id,
+          label: area.id,
+        })
+      })
+
+      // We start at a further zoom order to get info on all areas,
+      // so we need to zoom back to our intended default once those are gotten
+      map.easeTo({ zoom: 12, duration: 1500 })
```

Comment:
My hacky way of getting all the features in the source. When running `querySourceFeatures()` it only queries the features in the current viewport. So our default zoom made it so we would only know about ~50 features to start.

With this strategy, we start zoomed out to see all of Chicago, get all of our features, and then quickly zoom in to the zoom we actually wanted to start at.

In future projects, I'd probably lean towards keeping the geojson for things like this stored locally if possible. That way we don't have to wait for a response.

Reply:
Turns out using the `sourcedata` event had the same issue 😔